src/_pytest/_code/code.py

```diff
@@ -1221,7 +1221,7 @@ def getfslineno(obj: object) -> Tuple[Union[str, Path], int]:
         except TypeError:
             return "", -1
 
-        fspath = fn and absolutepath(fn) or ""
```

Comment:
it seems that this one confuses mypy and has to be typed as `Union[str, Path]`

Reply:
@withshubh I think it'd still be good to have this, can you try whether `fspath: Union[str, Path] = absolutepath(fn) if fn else ""` makes things work?